src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs

```diff
@@ -0,0 +1,201 @@
+
+                if (!IsSdkStyleProject || string.IsNullOrWhiteSpace(targetFramework) && targetFrameworks == null)
+                {
+                    //For non SDK style project, we cannot evaluate more info. Also there is no indication, whether the project
+                    //was restored or not, so it is not checked.
+                    _logger?.LogDebug("Project is non-SDK style, cannot evaluate restore status, succeeding.");
+                    return NonSDKStyleEvaluationResult.CreateSuccess(projectPath, evaluatedProject);
+                }
+
+                //For SDK-style project, if the project was restored "RestoreSuccess" property will be set to true.
+                if (!evaluatedProject.GetProperty("RestoreSuccess")?.EvaluatedValue.Equals("true", StringComparison.OrdinalIgnoreCase) ?? true)
+                {
+                    _logger?.LogDebug("Project is not restored, exiting.");
+                    return MSBuildEvaluationResult.CreateNoRestore(projectPath);
+                }
+
+                //If target framework is set, no further evaluation is needed.
+                if (!string.IsNullOrWhiteSpace(targetFramework))
+                {
+                    _logger?.LogDebug("Project is SDK style, single TFM:{0}, evaluation succeeded.", targetFramework);
+                    return SDKStyleEvaluationResult.CreateSuccess(projectPath, targetFramework, evaluatedProject);
+                }
+
+                //If target framework is not set, then presumably it is multi-target project.
+                //If there are no target frameworks, it is not expected.
+                if (targetFrameworks == null)
+                {
+                    _logger?.LogDebug("Project is SDK style, but does not specify the framework.");
+                    return MSBuildEvaluationResult.CreateFailure(projectPath, string.Format(LocalizableStrings.MSBuildEvaluator_Error_NoTargetFramework, projectPath));
+                }
+
+                //For multi-target project, we need to do additional evaluation for each target framework.
+                Dictionary<string, Project?> evaluatedTfmBasedProjects = new Dictionary<string, Project?>();
+                foreach (string tfm in targetFrameworks)
+                {
+                    _logger?.LogDebug("Evaluating project for target framework: {0}", tfm);
+                    evaluatedTfmBasedProjects[tfm] = RunEvaluate(projectPath, tfm);
+                }
+                _logger?.LogDebug("Project is SDK style, multi-target, evaluation succeeded.");
+                return MultiTargetEvaluationResult.CreateSuccess(projectPath, evaluatedProject, evaluatedTfmBasedProjects);
+
+            }
+            catch (Exception e)
+            {
+                _logger?.LogDebug("Unexpected error: {0}", e.Message);
+                return MSBuildEvaluationResult.CreateFailure(projectPath, e.Message);
```

Comment:
Using `Exception.Message` is generally a code smell, because you lose valuable context like the stack trace. Maybe log the full `e.ToString()` but put only the message in the result, or vice versa?

Reply:
Message in result may be shown to user. Stack trace is not shown to user, unless in verbose mode, so adjusted debug message only.